pkgs/by-name/ki/kimai/package.nix

```diff
@@ -0,0 +1,61 @@
+{pkgs ? import <nixpkgs> {
+  inherit system;
+}, system ? builtins.currentSystem,
+config,
+lib,
+consoleCmd ? "kimai-console",
+databaseUrl ? "",
+mailerFrom ? "",
+mailerUrl ? "",
+appSecret ? "",
+corsAllowOrigin ? "",
+...
+}:
+
+let
+  # userCfg = config.services.kimai;
+  phpPackage = import ./composer2nix/default.nix {
+    inherit pkgs system lib;
+  };
+in
+phpPackage.override {
+  postInstall = ''
+    # ensure the dirs exist, just in case,
+    # move any initialized data to "sample" to make space for symlinks.
+    # Finally, symlink to a directory outside of the nix store so that kimai
+    # may write to them without issue.
+    # These symlinks are created in the nix store but their targets need to be created later on,
+    # when the service starts.
+
+    mkdir -p ./var/cache
+    mkdir -p ./var/log
+    mkdir -p ./var/data
+    mv ./var/cache ./var/cache.sample
+    mv ./var/log ./var/log.sample
+    mv ./var/data ./var/data.sample
+    ln -s /var/lib/kimai/var/cache ./var/cache
+    ln -s /var/lib/kimai/var/log ./var/log
+    ln -s /var/lib/kimai/var/data ./var/data
+    ln -s /var/lib/kimai/var/plugins ./var/plugins
+
+    # remove cache on installation because sometimes references to previous builds (nix paths) can
+    # be left behind in this dir
+    rm -rf /var/lib/kimai/var/cache
+
+    # Create the .env file
+    # https://github.com/kimai/kimai/blob/main/.env.dist
+    rm -rf ./.env
+    touch ./.env
+    echo -e 'DATABASE_URL=${databaseUrl}\n' >> ./.env
+    echo -e 'MAILER_FROM=${mailerFrom}\n' >> ./.env
+    echo -e 'MAILER_URL=${mailerUrl}\n' >> ./.env
+    echo -e 'APP_SECRET=${appSecret}\n' >> ./.env
+    echo -e 'CORS_ALLOW_ORIGIN=${corsAllowOrigin}\n' >> ./.env
```

Comment:
I tried passing these as env vars to the `services.phpfpm.pools.${kimaiName}` systemd service, but PHP just refused to parse the values without failing.

Reply:
These work fine in the `.env` file though :)

Reply:
This is a less than ideal way to configure a package, overriding shouldn't be necessary as it'll cause a rebuild.

Instead maybe look at making the module create a directory that mounts the sources and then generates a `.env` file.